src/main/java/ua/chernonog/onlinebookstore/repository/book/spec/TitleSpecificationProvider.java

```diff
@@ -0,0 +1,21 @@
+package ua.chernonog.onlinebookstore.repository.book.spec;
+
+import java.util.Arrays;
+import org.springframework.data.jpa.domain.Specification;
+import org.springframework.stereotype.Component;
+import ua.chernonog.onlinebookstore.entity.Book;
+import ua.chernonog.onlinebookstore.repository.SpecificationProvider;
+
+@Component
+public class TitleSpecificationProvider implements SpecificationProvider<Book> {
+    @Override
+    public String getKey() {
+        return "title";
+    }
+
+    public Specification<Book> getSpecification(String[] params) {
+        return (root, query, criteriaBuilder) ->
+                root.get("title")
```

Comment:
Code duplication, no constants

Reply:
Actually strings 'title' and 'author' is used in several classes, so I suggest creating class where you will keep constants and reuse them when it is required